apis/Google.Cloud.BigQuery.V2/Google.Cloud.BigQuery.V2.IntegrationTests/JobsTest.cs

```diff
@@ -67,5 +69,210 @@ public void ListJobs_FullProjection()
+        }
+
+        [Fact]
+        public void LoadJob_Labels()
+        {
+            var bqClient = BigQueryClient.Create(_fixture.ProjectId);
+            var originTable = bqClient.GetTable(_fixture.DatasetId, _fixture.HighScoreTableId);
+            var destinationBucket = _fixture.StorageBucketName;
+            var destinationObject = _fixture.GenerateStorageObjectName();
+            var destinationUri = $"gs://{destinationBucket}/{destinationObject}";
+            var destinationReference = bqClient.GetTableReference(_fixture.DatasetId, _fixture.CreateTableId());
+
+            // Just extracting the data into GCS
+            var extractJob = originTable.CreateExtractJob(destinationUri);
+            extractJob = extractJob.PollUntilCompleted().ThrowOnAnyError();
+
+            var loadJob = bqClient.CreateLoadJob(
+                destinationUri, destinationReference, originTable.Schema, 
+                new CreateLoadJobOptions() { SkipLeadingRows = 1, Labels = JobLabels });
+            VerifyJobLabels(loadJob?.Resource?.Configuration?.Labels);
+
+            loadJob = loadJob.PollUntilCompleted().ThrowOnAnyError();
+            VerifyJobLabels(loadJob?.Resource?.Configuration?.Labels);
+        }
+
+        [Fact]
+        public void JobLabels_InvalidCharactersKey()
+        {
+            var client = BigQueryClient.Create(_fixture.ProjectId);
+            var table = client.GetTable(_fixture.ProjectId, _fixture.DatasetId, _fixture.HighScoreTableId);
+            var options = new QueryOptions()
+            {
+                Labels = new Dictionary<string, string>()
+                {
+                    {"This is Invalid & throws", "this_is_valid-2" }
+                }
+            };
+
+            Assert.ThrowsAny<GoogleApiException>(() => client.CreateQueryJob($"SELECT * FROM {table}", null, options));
+        }
+
+        [Fact]
+        public void JobLabels_InvalidCharactersValue()
+        {
+            var client = BigQueryClient.Create(_fixture.ProjectId);
+            var table = client.GetTable(_fixture.ProjectId, _fixture.DatasetId, _fixture.HighScoreTableId);
+            var options = new QueryOptions()
+            {
+                Labels = new Dictionary<string, string>()
+                {
+                    {"this_is_valid-2", "This is Invalid & throws" }
+                }
+            };
+
+            Assert.ThrowsAny<GoogleApiException>(() => client.CreateQueryJob($"SELECT * FROM {table}", null, options));
+        }
+
+        internal static IDictionary<string, string> JobLabels =>
+            new Dictionary<string, string>()
+            {
+                // This one is a little absurd for jobs because labels can only be set
+                // on job creation. This means not to set the label because this is copying
+                // the bahaviour of labels for other resources like DataSets and Table.
```

Comment:
Nit: bahaviour => behaviour (or behavior if we're going for US spelling; I don't mind either way)

Reply:
Done